test/unit/cryptoTest.ts

```diff
@@ -130,5 +133,14 @@ describe('CryptoTest', function() {
       secret = crypto.encrypt(undefined);
       expect(secret).to.equal(undefined);
     });
+
+    it('Decrypt should fail, error handling', () => {
+      const sandbox = sinon.createSandbox();
+      const message = Messages.loadMessages('@salesforce/core', 'crypto').getMessage('MacKeychainOutOfSync');
+      const errorMessage = SfdxError.wrap(new Error(message));
+      sandbox.stub(Crypto.prototype, 'decrypt').throws(errorMessage);
+      expect(() => Crypto.prototype.decrypt.call(secret)).to.throw(errorMessage);
+      sandbox.restore();
+    });
```

Comment:
You'll have to do something like this instead:
`const errMsg = 'decrypt failed';
stubMethod($$.SANDBOX, os, 'platform').returns('darwin');
stubMethod($$.SANDBOX, crypto, 'createDecipheriv').callsFake(() =>
      ({
        setAuthTag: () => throw new Error(errMsg),
        update: () => {},
        final: () => {}
      })
    );
crypto = new Crypto();
expect(crypto.decrypt('testing')).to.throw() // check that the error message is there`

Also would be worth adding another test with the env var set to make sure the error message isn't added.